api/controllers/calendarEntry.controller.ts

```diff
@@ -180,14 +145,73 @@ export const createCalendarEntry = async (
 };
 
 export const getCalendarEntries = async (
-  _req: Request,
+  req: Request,
   res: Response,
   _next: NextFunction,
 ) => {
   try {
-    const entries: CalendarEntry = await CalendarEntry.find();
-    res.status(200).json(entries);
+    const { start, end } = req.query;
+    const nonRecurringEntries: NonRecurringEntry[] = await CalendarEntry.find()
+      .where("recurring")
+      .equals(false)
+      .where("startTimeUtc")
+      .gte(start)
+      .where("endTimeUtc")
+      .lte(end);
+
+    const recurringEntriesWithinDateRange: RecurringEntry[] =
+      await CalendarEntry.find()
+        .where("recurring")
+        .equals(true)
+        .where("startTimeUtc")
+        .gte(start)
+        .where("recurrenceEndsUtc")
+        .lte(end);
+
+    const recurringEntriesBeginsBeforeEndsWithin: RecurringEntry[] =
+      await CalendarEntry.find()
+        .where("recurring")
+        .equals(true)
+        .where("startTimeUtc")
+        .lt(start)
+        .where("recurrenceEndsUtc")
+        .lte(end);
+
+    const recurringEntriesBeginsWithinEndsAfter: RecurringEntry[] =
+      await CalendarEntry.find()
+        .where("recurring")
+        .equals(true)
+        .where("startTimeUtc")
+        .gte(start)
+        .where("recurrenceEndsUtc")
+        .gte(end);
+
+    const recurringEntriesBeginsBeforeEndsAfter: RecurringEntry[] =
+      await CalendarEntry.find()
+        .where("recurring")
+        .equals(true)
+        .where("startTimeUtc")
+        .lt(start)
+        .where("recurrenceEndsUtc")
+        .gt(end);
```

Comment:
I don't think we need to separate them by when they end. These 4 queries can be combined into a single query because the treatment of the data is the same in either case:
```suggestion
    const recurringEntriesBeginsWithin: RecurringEntry[] =
      await CalendarEntry.find()
        .where("recurring")
        .equals(true)
        .where("startTimeUtc")
        .gte(start);

    const recurringEntriesBeginsBefore: RecurringEntry[] =
      await CalendarEntry.find()
        .where("recurring")
        .equals(true)
        .where("startTimeUtc")
        .lt(start)
        .where("endTimeUtc")
        .gt(start);
```